utils/server/allocatedWorkers.js

```diff
@@ -34,13 +35,16 @@ export const deleteResidentAllocatedWorker = async (id, body) => {
   return { test: 'test' };
 };
 
-export const addAllocatedWorker = async (resident_id, body) => {
-  const { data } = await axios.post(
-    `${ENDPOINT_API}/allocations`,
-    { resident_id, ...body },
-    {
-      headers: { 'Content-Type': 'application/json', 'x-api-key': AWS_KEY },
-    }
-  );
+const addAllocatedWorkerSchema = yup.object().shape({
+  mosaicId: yup.number().required().integer(),
+  allocatedWorkerId: yup.number().required().integer(),
+  allocatedBy: yup.string().email().required(),
+});
+
+export const addAllocatedWorker = async (mosaicId, params) => {
+  const body = await addAllocatedWorkerSchema.validate({ mosaicId, ...params });
```

Comment:
Never seen `yup` before, just confirming this line validates if the request body matches the defined schema above correct?

Reply:
Yes, in this case it's solving two problems, validation and casting of the values.
The BE is expecting `ids` as number, but we are getting back as `string`, `yup` is checking that the string contains a number and if it's so it is transforming it in an integer, making the BE happy.
As we are only using it in the BE side we shouldn't worry about the size of the lib, so I thought it was a nice to have and a good feedback for the devs.